File: go-service/models/role.go
package models

import "time"

// Role represents a user role in the system
type Role struct {
	ID          int        `db:"Id" json:"id"`
	Name        string     `db:"Name" json:"name"`
	Description *string    `db:"Description" json:"description,omitempty"`
	CreatedAt   time.Time  `db:"CreatedAt" json:"createdAt"`
	UpdatedAt   *time.Time `db:"UpdatedAt" json:"updatedAt,omitempty"`
}

File: frontend/src/app/core/domain/role/role.ts
import { User } from "../domain.barrel";

/**
 * Represents a Role
 */
export interface Role {
  /** ID */
  id: number;

  /** Name - role name */
  name: string;

  /** Description - role description */
  description?: string | null;

  /** Users - users with this role */
  users?: User[];

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;

  /** Deleted At */
  deletedAt?: Date | null;
}
